Add date range arguments to admin UI for generating data.

rawdreeg opened the pull request:
### All Submissions:

* [x] Have you followed the [Contributing guidelines](https://github.com/woocommerce/wc-smooth-generator/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/wc-smooth-generator/pulls) for the same update/change?

### Changes proposed in this Pull Request:

While generating some orders to test some reports, I realised that all orders generated via the UI have the same date. @garymurray has brought this up before, so I decided to implement it

This PR adds dates fields to the form and provides `start-date` and `end-date` args to the `generate orders` and `generate products` jobs.

### How to test the changes in this Pull Request:

![image](https://private-user-images.githubusercontent.com/4209011/424467438-b695c9dd-9778-40bf-8083-7fc5ef55c105.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzOTM1ODYsIm5iZiI6MTc0MjM5MzI4NiwicGF0aCI6Ii80MjA5MDExLzQyNDQ2NzQzOC1iNjk1YzlkZC05Nzc4LTQwYmYtODA4My03ZmM1ZWY1NWMxMDUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTlUMTQwODA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTNjZTQwZDQ3ZmMxMDk1ZjQ1ZDU1ZTQ0NjBjMmYwNmQwZTYwZDBjOGYzNTUzMzgxMGVlNDBkZmY5MDM5NzFkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-yCvrp72_kP9WxlgI5sEwjx7HikY4tJNnd3Slab_3aQ)

1. Using a build from this branch
2. Navigate to WP admin -> Tools -> Smooth Generator. Notice the new date input in the generate orders form.
3. Confirm that this new parameter works by generating a few orders with a start date different from the current date. This should generate orders with various date between your selected start date and the current date.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Add -  Add start date to admin UI for generating orders.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

rawdreeg committed:
Add start date argument to admin UI for generating orders.

budzanowski commented:
What happens to products? Will they be generated on the same date? I am just think that products can be created after orders that are using them.

mikkamp commented:
Just a fly by comment.

According to the docs for the CLI commands:

> Generate orders with random dates between --date-start and the current date.
> `wp wc generate orders <nr of orders> --date-start=2018-04-01`

Whereas from the UI it looks like we will be generating orders on a specific day. Should the `date-end` also be possible?

rawdreeg commented:
> What happens to products? Will they be generated on the same date? I just think that products can be created after orders that are using them.

Thanks @budzanowski. That's correct. However, it is also possible to generate orders without products. So we haven't been enforcing accuracy on these scenarios, lol. Applying the date range argument to both products and orders should help here.

rawdreeg commented:
Thanks @mikkamp,

> Whereas from the UI it looks like we will be generating orders on a specific day. Should the date-end also be possible?

I think if you don't specify a date-end, the current date is assumed to be the end date. But I agree we can add it as an option.

mikkamp commented:
> I think if you don't specify a date-end, the current date is assumed to be the end date. But I agree we can add it as an option.

Yes it will use the current date. I mainly meant to imply that the UI just has a single date picker, it doesn't specify what that date signifies, so my visual interpretation would be that it would generate 10 orders for the date that I select. But instead it uses random dates for a date range where I was able to select the start date. So even just a small blurb of text explaining what the date does is enough.

rawdreeg committed:
Implement full date range and apply arguments to products and orders.

rawdreeg commented:
@budzanowski

I reworked the PR to apply the date argument to products as well.

![Screen Recording 2025-03-19 at 14 09 40](https://github.com/user-attachments/assets/b695c9dd-9778-40bf-8083-7fc5ef55c105)

>Yes it will use the current date. I mainly meant to imply that the UI just has a single date picker, it doesn't specify what that date signifies

 @mikkamp, that makes sense. I added the end-date option too.


I wonder whether I should put the date options before the "generate" fields on the form.

budzanowski commented:
> I wonder whether I should put the date options before the "generate" fields on the form.

I don't think that it is necessary.
